src/irmin-pack/ext.ml

```diff
@@ -187,12 +183,64 @@ module Maker (Config : Conf.S) = struct
                   Lwt.fail e
               | e -> Lwt.fail e)
 
-        (** Stores share instances in memory, one sync is enough. *)
-        let sync t = Contents.CA.sync (contents_t t)
+        (** For crash consistency and SWMR consistency, the order in which we
+            [flush], [sync] and [clear] the various files should carefuly chosen
+            in order to avoid introducing dangling references from file to file.
+
+            There are 4 types of files (called components below), the ones in
+            index, the dict, the pack file and the branch one. Here are the
+            relations between them:
+
+            - the branch store references commit hashes in index,
+            - index references pack entries,
+            - the pack file references dict entries and
+            - the dict references nothing.
+
+            In case of crash (or an ro instance that concurrently syncs), the
+            [clear] the order should be: branch, index, pack and dict.
```

Comment:
What is `the [clear] the order` ? Do you mean: in order to avoid problems when the system crashes, the order should be: ...? And then, shouldn't we flush the dict first, then the pack, etc?

Reply:
BTW I really appreciate that you tried to clarify things with this documentation. I think we really need more documentation, especially in areas where there is "tricky" code.